### PR TITLE
Add Component for Non-Existent Routes

### LINKED ISSUE #35: 22. As a user, I want to be informed if I enter a page that doesn’t exist
## Summary:
To enhance the user experience when they enter a non-existent page, show a message to the user indicating that the page is not found and provide a link that will allow them to return to the home page.

## Acceptance criteria:

- [ ] Display a clear message indicating that the requested page could not be found.
- [ ] Provide a link to navigate back to the home page.

## Notes:

- Create a NotFoundPage component.
- Add a Route that handles a 404 page to the Routes.

### PR DESCRIPTION
**Pull request by trushmi**
## Description
This code adds functionality that allows users, when they navigate to a non-existent route in the app, to now see a message explaining that the page does not exist, and they have a link to navigate back to the home page.

## Related Issue

Closes #35

## Acceptance Criteria

- [ ]  Display a clear message indicating that the requested page could not be found.
- [ ]  Provide a link to navigate back to the home page.

## Type of Changes

enhancement

### Before

Screenshot 2024-03-26 at 10 49 20 AM

### After
Screenshot 2024-03-26 at 10 53 16 AM

## Testing Steps / QA Criteria

- git pull
- git checkout it-page-not-found
- npm ci
- npm start
- Navigate to the non-existent route in the app and check if you can see a message that explains that this page doesn't exist. 
- Test the link to see if it navigates to the home page.

### CODE REVIEW

**Review by sillytsundere — APPROVED** [review]
This looks great! and everything worked for me when i pulled and demo-ed! Great job!

**Review by rachelspencer — APPROVED** [review]
Nice job Iryna! This adds a nice touch to the app!

File: src/components/NotFoundPage.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (timothy-taylor) [inline src/components/NotFoundPage.jsx:1]: I don't think you need to import React in vite projects if you are not actually using the variable, but not a big deal.

**Review by timothy-taylor — APPROVED** [review]
looks good! 🚀

**Pull request merged**